src/Symfony/Bridge/Doctrine/Tests/Types/DatePointTypeTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Tests\Types;
+
+use DateTimeImmutable;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\MariaDBPlatform;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQLPlatform;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\Doctrine\Types\DatePointType;
+use Symfony\Component\Clock\DatePoint;
+
+final class DatePointTypeTest extends TestCase
+{
+    private DatePointType $type;
+
+    public static function setUpBeforeClass(): void
+    {
+        $name = DatePointType::NAME;
+        if (Type::hasType($name)) {
+            Type::overrideType($name, DatePointType::class);
+        } else {
+            Type::addType($name, DatePointType::class);
+        }
+    }
+
+    protected function setUp(): void
+    {
+        $this->type = Type::getType(DatePointType::NAME);
+        if (!class_exists(DatePoint::class)) {
+            self::markTestSkipped('The DatePoint class is not available.');
+        }
+    }
+
+    public function testDatePointConvertsToDatabaseValue()
+    {
+        $datePoint = new DatePoint('2025-03-03 12:13:14');
+
+        $expected = $datePoint->format('Y-m-d H:i:s');
+        $actual = $this->type->convertToDatabaseValue($datePoint, new PostgreSQLPlatform());
+
+        $this->assertEquals($expected, $actual);
```

Comment:
`assertSame()` should be used instead of `assertEquals()` where possible

Reply:
That makes sense when comparing objects, right?
There are many `assertEquals` when comparing scalars in the test suite (126 only in the Doctrine bridge, 6825 in all Symfony)

Reply:
This also makes sense when comparing scalar to ensure types are respected and that no type coercion is done. There are indeed many `assertEquals()` in the codebase and going through all of them would be tedious (thus we don't do it), but we try to be consistent where possible in the new code and use `assertSame()` everywhere it's possible (sometimes it's not). 🙂